### ISSUE TITLE
Add a vignette to document the options

### ISSUE THREAD

**Issue by janschulz** [IRkernel/IRkernel#278]
Mainly it should tell an enduser what plotting options there are (size of the plot,..., remove svg)

**Comment by flying-sheep** [comment]
why not a sub-website? i prefer not to download PDFs if i can read things online

**Comment by JanSchulz** [comment]
Because I go to CRAN (or `vignette(...)`) to find my documentation :-) And vignettes are nowadays html files: https://cran.r-project.org/web/packages/cowplot/vignettes/axis_position.html

They are rmarkdown files, so you could even compile them to a html file on our website.

**Comment by JanSchulz** [comment]
This is never going to fly on cran... :-/ What we could do is a nbconvert template which would output a rmarkdown document...

**Comment by flying-sheep** [comment]
bioconductor needed like a month to get it done.

**Comment by JanSchulz** [comment]
Reading tweets like https://twitter.com/xieyihui/status/590406807329222656 (from the knitr author!) makes me scared to submit anything to cran...

**Comment by JanSchulz** [comment]
[As a sidenote: I don't like the notebook as a "text editor"... That's why I ported knitr to python or better to the jupyter infrastructure... (-> [knitpy][https://github.com/JanSchulz/knitpy])]

**Comment by flying-sheep** [comment]
> Reading tweets like https://twitter.com/xieyihui/status/590406807329222656 (from the knitr author!) makes me scared to submit anything to cran...

well, you have to have the stamina to get through all the review rounds (which are as pedantic as my code style remarks, but i’m less insistent)

but if you resubmit often enough it’ll get through.

**Comment by JanSchulz** [comment]
> but if you resubmit often enough it’ll get through.

wanna start with repr? :-)

**Comment by janschulz** [comment]
Ok, I think this can be closed because it would already get a help file on CRAN.

But:
```
?IRkernel::jupyter_option_defaults
Error: c:/data/external/R/IRkernel/man/IRkernel.Rd:12-16: Tag \code not expected in code block
Traceback:

1. prepare_mimebundle(obj, .self$handle_display_error)
2. repr_text(obj)
3. repr_text.help_files_with_topic(obj)
4. repr_help_files_with_topic_generic(obj, tools::Rd2txt)
5. capture.output(Rd2_(rd, package = pkgname, outputEncoding = "UTF-8"))
6. evalVis(expr)
7. withVisible(eval(expr, pf))
8. eval(expr, pf)
9. eval(expr, envir, enclos)
10. Rd2_(rd, package = pkgname, outputEncoding = "UTF-8")
11. writeSection(Rd[[i]], sections[i])
12. writeCodeBlock(section, tag)
13. stopRd(block, Rdfile, "Tag ", tag, " not expected in code block")
14. stop(msg, call. = FALSE, domain = NA)
```

-> We should fix our docs
-> the repr method for that should catch such errors and not send a traceback?

**Comment by flying-sheep** [comment]
docs fixed in 7bbba4a6

**Comment by janschulz** [comment]
> i think the repr method should propagate errors. broken docs are broken, why paper over it?

In this case it should just send a shorter message because no one benefits from seeing our call stack... It makes a lot more sense to output the name instead and maybe what to do (report it in the package).
